sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1038,10 +1038,10 @@ private boolean isManagementToken(String lockToken) {
                 if (receiverOptions.isEnableAutoComplete() && throwable instanceof AmqpException) {
                     switch (dispositionStatus) {
                         case COMPLETED:
-                            return new ServiceBusAmqpException((AmqpException) throwable,
+                            return new ServiceBusReceiverException((AmqpException) throwable,
                                 ServiceBusErrorSource.COMPLETE);
                         case ABANDONED:
-                            return new ServiceBusAmqpException((AmqpException) throwable,
+                            return new ServiceBusReceiverException((AmqpException) throwable,
                                 ServiceBusErrorSource.ABANDONED);
                         default:
                             // Do nothing
```

Comment:
If the exception type is AmqpException then we should wrap it even in default case with UNKNOWN source.